### ISSUE TITLE
AS3 override not getting applied after CIS reboot/recreate in CIS 2.0

### ISSUE THREAD

**Issue by chen23** [F5Networks/k8s-bigip-ctlr#1351]
#### Setup Details
 CIS Version : 2.0.0
 Build: f5networks/k8s-bigip-ctlr:2.0.0       
 BIGIP Version: Big IP 15.1      
 AS3 Version: 3.20
 Agent Mode: AS3       
 Orchestration: OCP     
 Orchestration Version: 4.3       
 Pool Mode: Cluster      
 Additional Setup details: OVS

#### Description  
  changing from CIS 1.1.4.0 to CIS 2.0.0 an existing AS3 override no longer loads.

#### Steps To Reproduce
1) deploy CIS 1.14 
2) see override
3) deploy CIS 2.0.0 see it disappear

#### Expected Result 
it works

#### Actual Result  
it does not

#### Diagnostic Information

CIS 1.14
![image](https://user-images.githubusercontent.com/63689/84310605-e90d5880-ab2f-11ea-9123-e2ec94a54fa1.png)

CIS 2.0.0
![image](https://user-images.githubusercontent.com/63689/84310656-ffb3af80-ab2f-11ea-8fa4-ec8557f63c5b.png)

config (using operator)
```
apiVersion: cis.f5.com/v1
kind: F5BigIpCtlr
metadata:
  creationTimestamp: '2020-06-10T15:52:49Z'
  finalizers:
    - uninstall-helm-release
  generation: 5
  name: f5-server
  namespace: openshift-operators
  resourceVersion: '222009'
  selfLink: /apis/cis.f5.com/v1/namespaces/openshift-operators/f5bigipctlrs/f5-server
  uid: 43f83e45-f5e8-4f9f-9400-5ee155e1cca1
spec:
  args:
    agent: as3
    bigip_partition: ocp
    bigip_url: 10.1.20.240
    insecure: true
    log_as3_response: true
    log_level: DEBUG
    manage_routes: true
    namespace: default
    openshift_sdn_name: /Common/openshift_vxlan
    override_as3_declaration: default/f5-as3-declaration
    pool-member-type: cluster
    route_vserver_addr: 10.1.10.11
  bigip_login_secret: bigip-login
  image:
    pullPolicy: IfNotPresent
    repo: k8s-bigip-ctlr
    user: f5networks
  namespace: kube-system
  rbac:
    create: true
  resources: {}
  serviceAccount:
    create: true
    name: null
  version: 2.0.0

```

#### Observations (if any)

Also tried https://github.com/F5Networks/k8s-bigip-ctlr/blob/2.0-stable/docs/_static/config_examples/configmap/override-configmap/README.md and had same result.  Also observed that the AS3 declaration changes the name of object from "ingress_10_1_10_12_80" to "Ingress_10_1_10_12_80"

**Comment by strantalis** [comment]
I noticed this issue as well. I thought it was working but I think thats when we were on 1.14 but when we upgraded to 2.0 I am not seeing the overrides anymore.

**Comment by mdditt2000** [comment]
# AS3 override functionality
- "--override-as3-declaration=default/f5-as3-declaration"
 
No longer works in CIS 2.0 – Github https://github.com/F5Networks/k8s-bigip-ctlr/issues/1351
 
Files to re-create
 
https://github.com/mdditt2000/kubernetes-1-18/tree/master/cis%202.0/github-sr-issues/#1351

**Comment by mdditt2000** [comment]
@chen23 @strantalis please make sure that if you specify ingress or routes = true and using override you also add configmap = true

Example i set ingress=true and specified everything else to false. However that way override wont work. By default the values are set to true. So i changes --manage-configmaps=true and its working. 
```
args:
            - "--manage-ingress=true".  --- set to true for me
            - "--manage-routes=false"
            - "--manage-configmaps=true". --- need for configmap or remove argument as default is true
            - "--custom-resource-mode=false"
            # AS3 override functionality
            - "--override-as3-declaration=default/f5-as3-declaration"
```

Please let me know if you still having issue. Multiple dev folks validated that AS3 override is working fine.

**Comment by chen23** [comment]
it is working now.  in my previous configmap I had the labels indented incorrectly.  Is the labels change in the main documentation or only in the page that I referenced earlier?

**Comment by strantalis** [comment]
I think I have steps that can reproduce the problem I am seeing.

1. Initially apply the as3 override declaration while the CIS pod is running.(I noticed this does apply the override)
2. Restart the CIS pod. (After this restart it doesn't seem like CIS uses the override anymore and wipes out those changes). 
3. Then if you want delete the configmap and recreate it. CIS should apply those override changes again.

**Comment by mdditt2000** [comment]
@chen23 created Jira CONTCNTR-1896 for @vtrippel to update the configmap documentation

**Comment by mdditt2000** [comment]
@strantalis please use CIS 2.0 for testing. CIS 1.14 will require a restart the CIS pod when making a change to the as3 override declaration

**Comment by strantalis** [comment]
I am running 2.0.

**Comment by mdditt2000** [comment]
@strantalis testing with the following configuration 

https://github.com/mdditt2000/kubernetes-1-18/tree/master/cis%202.0/github-sr-issues/%231351

Let me see if i see the same as you. Standby

**Comment by mdditt2000** [comment]
AS3 override is working correctly

1) Improvements in documentation is required. Created Jira CONTCNTR-1896 and assigned to Tech writer

**Comment by mdditt2000** [comment]
Documents updated. https://clouddocs.f5.com/containers/latest/userguide/config-map.html. Closing issue

**Comment by mdditt2000** [comment]
Re-opened issue:
When CIS 2.0 reloads the AS3 override is sometimes removed. This issue has being resolved and will be released in CIS 2.1 - Internal Jira CONTCNTR-1898 for PM tracking 

**Hotfix available** - please contact me if you need the hotfix until CIS 2.1

**Comment by mdditt2000** [comment]
Jira is closed and merged and will be released in CIS 2.1 middle of next month. Closing this issue out
